rust/connlib/snownet/src/node.rs

```diff
@@ -276,6 +276,15 @@ where
                 &mut self.pending_events,
             );
 
+            // Initiate a new WG session.
+            //
+            // We can have up to 8 concurrent WireGuard sessions in boringtun before the oldest one get overwritten.
+            // Also, whilst we are handshaking a new session, we won't send another handshake.
+            // Thus, even rapidly successive connection upserts should be handled just fine.
```

Comment:
```suggestion
            // Thus, even rapid successive connection upserts should be handled just fine.
```

This seems to contradict the PR description where we state we debounce this for 20 seconds.

Reply:
What we debounce is sending the handshake. The connection upsert happens regardless but it is a no-op otherwise (if ICE credentials are the same).